### ISSUE TITLE
Improve documentation relating to Mailgun?

### ISSUE THREAD

**Issue by jaylenw** [firefly-iii/firefly-iii#3523]
I am running Firefly III version 5.2.4.

**Description**

I have tested my Firefly-iii installation with Mailgun and so far haven't had any issues with Firefly-iii sending emails with Mailgun. In the documentation, we read the following:

https://docs.firefly-iii.org/advanced-installation/email#mailgun

"Mailgun is an API that can send emails. Their service allows for 10,000 free messages each month. To configure these, fill in the MAILGUN_DOMAIN and MAILGUN_SECRET values in your .env file. Please note that I haven't actually tested this."

May we remove the sentence that says, "Please note that I haven't actually tested this." if I can show proof it is working properly? I would like the end user to have more confidence in configuring Mailgun if they wanted to.

I read Mailgun's documenation, https://documentation.mailgun.com/en/latest/quickstart-sending.html#send-with-smtp-or-api, to help understand and configure Firefly-iii to work with it. Here is my `.env` file documentation, https://github.com/jaylenw/firefly-iii-docker-setup#firefly-env. Below is the portion with Mailgun:

```
MAIL_DRIVER=mailgun # configured this because I am using Mailgun for emails
MAIL_FROM=<from-email-address> # this is important, set this using your verified Mailgun domain, example money@firefly.mydomain.com
MAILGUN_DOMAIN=<verified-mailgun-domain> # this is important, set your verified Mailgun domain, example firefly.mydomain.com
MAILGUN_SECRET=<your-mailgun-api-key>
MAILGUN_ENDPOINT=api.mailgun.net # this is fine if you are in the U.S.
```

Let me know how I can help. I appreciate your work. Thanks.

**Extra info**
 

Debug information generated at 2020-07-03 11:30:20 America/Los_Angeles for Firefly III version **5.2.4**.

| Variable | Content |
| --- | --- |
| FF version | 5.2.4 |
| FF API version | 1.1.0 |
| Installation ID | b2c27d92-be90-5c10-8589-005df5b314e6 |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | notice, stack |
| PHP version | 7.3.16 |
| Display errors | Off |
| Session start | 2020-07-01 00:00:00 |
| Session end | 2020-09-30 00:00:00 |
| Session first | 2020-05-11 00:00:00 |
| Error reporting | ALL errors |
| Host | Linux |
| Interface | apache2handler |
| UserID | 1 |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | pgsql |
| Login provider |   |
| Trusted proxies (.env) | ** |
| User agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.116 Safari/537.36 |

**Bonus points**

- I have read the FAQ at https://bit.ly/FF3-FAQ
- I added a screenshot
- I added log files 
- I was able to replicate the issue on the demo site.

**Comment by JC5** [comment]
Cool! Thanks for the compliments.

What you could do, is edit [this file](https://github.com/firefly-iii/docs/blob/main/new-docs/installation/email.md) and add a little bit about how you've configured Mailgun. You can remove the sentence and add your own notes. Once merged, the documentation will change for everybody to enjoy.

I can also do it for you, if you feel more confident about that :)

**Comment by JC5** [comment]
Oh and keep in mind that in the new release, the `MAIL_DRIVER` variable is called `MAIL_MAILER`.

**Comment by jaylenw** [comment]
Hello @JC5 , okay sounds good. Allow me to allocate some time to upgrade my local test environment to release, 5.3.0, and make the change from `MAIL_DRIVER` to `MAIL_MAILER`. Once I've tested that all is well with the new release with Mailgun, I will create a PR. Thanks.

**Comment by jaylenw** [comment]
Hello @JC5 , I have updated my local test environment to 5.3.0 and made the change from `MAIL_DRIVER` to `MAIL_MAILER`. Well I know the email works with Mailgun as I got an email that we caught an error in Firefly III :). See the screenshot and pdf attached. Let me proceed to make a PR for the Mailgun documentation.

![Screenshot from 2020-07-05 16-27-23](https://user-images.githubusercontent.com/6505395/86544596-1c4bc900-bedd-11ea-8192-0984c50758f6.png)


As far as the error I encountered, I will have to look more into this but it is related to https://github.com/firefly-iii/firefly-iii/blob/54dfb35007e9f69ec70cfafb2f77dc4a496686ad/app/Http/Controllers/Controller.php#L58

Is it that I am missing a key value pair in my `.env` file?
[Firefly III Exception _(.pdf](https://github.com/firefly-iii/firefly-iii/files/4875946/Firefly.III.Exception._.pdf)
